### ISSUE TITLE
How to properly fail a test?

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#7910]
In web frameworks there is often an error handler that catches internal errors. For example in Vue there is

```js
Vue.config.errorHandler = (err, vm, info) => {
  ...
}
```

How can we fail a running test if an error is caught this way? The only solution I found was to do the following

```js
// exclude these lines from code coverage
/* istanbul ignore next */
if (window.Cypress) {
  // send any errors caught by the Vue handler
  // to the Cypress top level error handler to fail the test
  Vue.config.errorHandler = window.top.onerror
}
```

But there has to be a better way

<img width="1297" alt="Screen Shot 2020-07-07 at 3 27 47 PM" src="https://user-images.githubusercontent.com/2212006/86832656-75933400-c066-11ea-887f-cec027dc05b1.png">

**Comment by jennifer-shehane** [comment]
You should just be able to throw the Error right? Because Cypress fails the test on any exceptions from the application.

**Comment by bahmutov** [comment]
So no, @jennifer-shehane, unfortunately, throwing the error does nothing in this case, since they were still intercepted by Vue's error handler. After struggling for a while on this with @flotwig found that Cypress's top window handler works.

For now, this is a limitation that is completely breaking first-time users, or even seasoned users of Cypress. Imagine you are testing a Vue app like this one. The test passes or not, does not matter - but you see Vue errors in the DevTools. Aren't you wondering - why the test is passing if there are errors? Ok, so you figure out the Vue error handler, but then you _still cannot make it fail the test_. So this is why I have opened this issue. We got to provide an official answer, and at least now I can link my descriptions of what to do in this case. Later on, if we implement a different way to fail a test, people will still find it.

I have added this by default to Vue component testing https://github.com/bahmutov/cypress-vue-unit-test/commit/604edfaf1d1784a7224adfd95dedc749a3de23e5
